tests/conftest.py

```diff
@@ -236,26 +236,70 @@ def _benchmark_memory(history):
             events_df["peak_measured_mem_end_mb"] = (
                 events_df["peak_measured_mem_end"] / 1_000_000
             )
-            
+
             test_run_benchmark.projected_memory = float(plan_df["projected_mem_mb"].max())
             test_run_benchmark.peak_memory = float(events_df["peak_measured_mem_end_mb"].max())
             test_run_benchmark.average_memory = float(events_df["peak_measured_mem_end_mb"].mean())
             
     yield _benchmark_memory
 
 
+
+@pytest.fixture(scope="function")
+def benchmark_tasks(test_run_benchmark):
+    """Benchmark the number and cumulative duration of tasks run.
+
+    Yields
+    ------
+    Context manager factory function which takes an instantiated cubed HistoryCallback object
+    as input. The context manager records number and cumulative duration of
+    tasks run while executing the ``with`` statement if run as part of a benchmark,
+    or does nothing otherwise.
+
+    Example
+    -------
+    .. code-block:: python
+
+        def test_something(benchmark_tasks):
+            history = cubed.extensions.history.HistoryCallback()
+            with benchmark_tasks(history):
+                cubed.compute(*arrs, callbacks=[history])
+    """
+
+    @contextlib.contextmanager
+    def _benchmark_tasks(history):
+        if not test_run_benchmark:
+            yield
+        else:
+            yield
+
+            import pandas as pd
+
+            # read off the memory values and write them into the database
+            plan_df = pd.DataFrame(history.plan)
+            events_df = pd.DataFrame(history.events)
```

Comment:
Seems a little silly to be duplicating this code to create the dataframe, but it's nice to be able to separate `benchmark_memory` from `benchmark_tasks` I guess.